transport/http2_client.go

```diff
@@ -158,10 +159,15 @@ func newHTTP2Client(addr string, opts *ConnectOptions) (_ ClientTransport, err e
 			return nil, ConnectionErrorf("transport: %v", err)
 		}
 	}
+	ua := primaryUA
+	if opts.UserAgent != "" {
+		ua += " " + opts.UserAgent
```

Comment:
This is the wrong way around. The higher level User-Agent string should be coming first. I pointed you at the RFC for that.